elastic/e2e-testing
# fix: consistently read versions in Ansible

## What does this PR do?
It fixes how the version variables are set in Ansible defining the following criterium:

0. if STACK_VERSION does not exist, use the .stack_version file
1. if the variable exists, use it (i.e. BEAT_VERSION, ELASTIC_AGENT_VERSION, KIBANA_VERSION)
2. if not, use STACK_VERSION
3. if not, use the .stack_version



## Why is it important?
Testing locally for versions that are different to the one in the .stack_version file is not possible, as the version is overwritten



## Checklist




## Author's Checklist


- [ ] Is the Jenkinsfile passing the variables to the stack/test execution? I know the variables are defined at the top-level, but want to make sure they are passed to the enclosing blocks to run the stack node and to run the tests in the test node.

## How to test this PR locally

1. Create a stack
```
make -C .ci create-stack STACK_VERSION="FOOO" BEAT_VERSION="FOOO"
```

The logs should display that the variable will be applied to the `.env` file



## Related issues


- Closes #2714
